substrate/zombienet/zombienet-sdk/tests/zombie_ci/block_building_warp_sync.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use crate::utils::{
+	initialize_network, BEEFY_BEST_BLOCK_METRIC, BEST_BLOCK_METRIC, DEFAULT_CHAIN_SPEC,
+	DEFAULT_DB_SNAPSHOT_URL, DEFAULT_SUBSTRATE_IMAGE,
+};
+use anyhow::{anyhow, Result};
+use env_logger::Env;
+use zombienet_orchestrator::network::node::LogLineCountOptions;
+use zombienet_sdk::{NetworkConfig, NetworkConfigBuilder, NetworkNode};
+
+const INTEGRATION_IMAGE_ENV: &str = "ZOMBIENET_INTEGRATION_TEST_IMAGE";
+const DB_SNAPSHOT_ENV: &str = "DB_SNAPSHOT";
+const CHAIN_SPEC_ENV: &str = "WARP_CHAIN_SPEC_PATH";
+const DB_BLOCK_HEIGHT_ENV: &str = "DB_BLOCK_HEIGHT";
+
+const ROLE_TIMEOUT_SECS: u64 = 60;
+const PEER_TIMEOUT_SECS: u64 = 60;
+const BOOTSTRAP_TIMEOUT_SECS: u64 = 180;
+const METRIC_TIMEOUT_SECS: u64 = 60;
+const NEW_BLOCK_TIMEOUT_SECS: u64 = 120;
+const LOG_TIMEOUT_LONG_SECS: u64 = 60;
+const LOG_TIMEOUT_SHORT_SECS: u64 = 10;
+const LOG_ERROR_TIMEOUT_SECS: u64 = 10;
+const BEEFY_SYNC_TIMEOUT_SECS: u64 = 180;
+const BEEFY_PROGRESS_TIMEOUT_SECS: u64 = 60;
+
+const PEERS_THRESHOLD: f64 = 2.0;
+const MIN_BOOTSTRAP_BLOCK: f64 = 1.0;
+
+const VALIDATORS: [&str; 2] = ["alice", "bob"];
+const FOLLOWERS: [&str; 2] = ["charlie", "dave"];
+
+#[tokio::test(flavor = "multi_thread")]
+async fn block_building_warp_sync() -> Result<()> {
+	let _ = env_logger::Builder::from_env(Env::default().default_filter_or("info")).try_init();
+
+	ensure_env_defaults();
+
+	log::info!("Spawning network");
+	let config = build_network_config()?;
+	let network = initialize_network(config).await?;
+
+	network.wait_until_is_up(BOOTSTRAP_TIMEOUT_SECS).await?;
+
+	check_node_roles(&network).await?;
+	check_peers(&network).await?;
+
+	let db_snapshot_height = resolve_db_snapshot_height(&network).await?;
+
+	verify_bootstrap_height(&network, db_snapshot_height).await?;
+	verify_new_blocks(&network, db_snapshot_height).await?;
+
+	let dave = network.get_node("dave")?;
+	verify_dave_progress(dave, db_snapshot_height).await?;
+	verify_dave_logs(dave).await?;
+	verify_dave_beefy(dave, db_snapshot_height).await?;
+	verify_dave_log_errors_absent(dave).await?;
```

Comment:
Why we are _naming_ this methods and not make them general and pass the node_name as arg?